src/basic/random-util.c

```diff
@@ -93,34 +93,31 @@ int acquire_random_bytes(void *p, size_t n, bool high_quality_required) {
 }
 
 void initialize_srand(void) {
-        static bool srand_called = false;
-        unsigned x;
+        static bool initialized = false;
+        unsigned x = 0;
 #ifdef HAVE_SYS_AUXV_H
-        void *auxv;
+        char *auxv;
 #endif
 
-        if (srand_called)
+        if (initialized)
                 return;
 
+        x ^= (unsigned) now(CLOCK_REALTIME);
+        x ^= (unsigned) gettid();
+
+        initialized = true;
+
 #ifdef HAVE_SYS_AUXV_H
         /* The kernel provides us with 16 bytes of entropy in auxv, so let's
          * try to make use of that to seed the pseudo-random generator. It's
          * better than nothing... */
 
-        auxv = (void*) getauxval(AT_RANDOM);
-        if (auxv) {
-                assert_cc(sizeof(x) < 16);
-                memcpy(&x, auxv, sizeof(x));
-        } else
+        auxv = (char*) getauxval(AT_RANDOM);
+        if (auxv)
+                (void) initstate(x, auxv, 16);
```

Comment:
I am not sure I grok initstate(). The array parameter, is that copied? I don't think it is, because then the parameter would be "const", but it isn't. Hence you are directly using the auxval array as state array in-place, that can't be right?

The docs for this are awful, but it appears to me, that initstate() is just a way to tell the libc to use some memory region for the RNG state, and the contents of it are not relevant. But I am not sure